packages/purpose-platformstate-writer/src/utils.ts

```diff
@@ -97,6 +131,403 @@ export const deletePlatformPurposeEntry = async (
+): Promise<void> => {
+  const runPaginatedUpdateQuery = async (
+    dynamoDBClient: DynamoDBClient,
+    purpose: Purpose,
+    purposeState: ItemState,
+    purposeVersionId: PurposeVersionId,
+    exclusiveStartKey?: Record<string, AttributeValue>
+  ): Promise<void> => {
+    const result = await readTokenEntriesByGSIPKPurposeId(
+      dynamoDBClient,
+      purpose.id,
+      exclusiveStartKey
+    );
+    const gsiPKConsumerIdEServiceId = makeGSIPKConsumerIdEServiceId({
+      consumerId: purpose.consumerId,
+      eserviceId: purpose.eserviceId,
+    });
+    const platformAgreementEntry =
+      await readPlatformAgreementEntryByGSIPKConsumerIdEServiceId(
+        dynamoDBClient,
+        gsiPKConsumerIdEServiceId
+      );
+    const catalogEntry = platformAgreementEntry
+      ? await readCatalogEntry(
+          dynamoDBClient,
+          makePlatformStatesEServiceDescriptorPK({
+            eserviceId: purpose.eserviceId,
+            descriptorId: unsafeBrandId<DescriptorId>(
+              platformAgreementEntry.agreementDescriptorId
+            ),
+          })
+        )
+      : undefined;
+
+    for (const entry of result.tokenStateEntries) {
+      const tokenEntryPK = entry.PK;
+      const isAgreementMissingInTokenTable =
+        platformAgreementEntry &&
+        (!entry.GSIPK_consumerId_eserviceId ||
+          !entry.agreementId ||
+          !entry.agreementState);
+
+      // Agreement data from platform-states
+      const agreementExpressionAttributeValues: Record<string, AttributeValue> =
+        isAgreementMissingInTokenTable
+          ? {
+              ":GSIPK_consumerId_eserviceId": {
+                S: platformAgreementEntry.GSIPK_consumerId_eserviceId,
+              },
+              ":agreementId": {
+                S: extractAgreementIdFromAgreementPK(platformAgreementEntry.PK),
+              },
+              ":agreementState": {
+                S: platformAgreementEntry.state,
+              },
+            }
+          : {};
+      const agreementUpdateExpression = isAgreementMissingInTokenTable
+        ? `, GSIPK_consumerId_eserviceId = :GSIPK_consumerId_eserviceId, 
+      agreementId = :agreementId, 
+      agreementState = :agreementState`
+        : "";
+
+      // Descriptor data from platform-states
+      const isDescriptorDataMissingInTokenTable =
+        platformAgreementEntry &&
+        catalogEntry &&
+        (!entry.GSIPK_eserviceId_descriptorId ||
+          !entry.descriptorAudience ||
+          !entry.descriptorState);
+
+      const descriptorExpressionAttributeValues: Record<
+        string,
+        AttributeValue
+      > = isDescriptorDataMissingInTokenTable
+        ? {
+            ":GSIPK_eserviceId_descriptorId": {
+              S: makeGSIPKEServiceIdDescriptorId({
+                eserviceId: purpose.eserviceId,
+                descriptorId: unsafeBrandId<DescriptorId>(
+                  platformAgreementEntry.agreementDescriptorId
+                ),
```

Comment:
is this already branded?
```suggestion
                descriptorId: platformAgreementEntry.agreementDescriptorId,
```

Reply:
Yes, it was. Removed. Thanks.